### ISSUE TITLE
[$20]Datepicker: The instructions for the screen reader users states 'Press the down arrow key to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.'. Nothing happens when we press the '?' on the keyboard.

### ISSUE THREAD

**Issue by veshu** [topcoder-platform/community-app#3231]
**Describe the bug**

Date picker:   The instructions for the screen reader users states 'Press the down arrow key   to interact with the calendar and select a date. Press the question mark key   to get the keyboard shortcuts for changing dates.'. Nothing happens when we   press the '?' on the keyboard.

**To Reproduce**

1. Go   to the Challenges page: https://www.topcoder.com/challenges.
 2. Ensure the   screen readr is switched on. 
3. Activate the filter.
 4. Press the tab key   till you arrive at the start date field. 
5. Press the '?' key from the   keyboard as mentioned in the instructions.
6. Notice that no changes are   announced by the screen readers.

**Expected behavior**

There   should be button 'callendar keyboard shortcuts' which when activated should   show the keyboard shortcuts. Hence the instructions given are not clear.

**Actual behavior**

Screen   reader doesn't announce any keyboard shortcuts to work with the callendar.

**Additional context**
Type: Challenge   Filter Box

Page: https://www.topcoder.com/challenges

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30099900 has been created for this ticket.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30099900 has been updated - the new changes has been updated for this ticket.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30099900 has been updated - it has been assigned to rashmi73.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by rashmi73** [comment]
@veshu PR soon in few hour

**Comment by r0hit-gupta** [comment]
@rashmi73 are you still working on it?

**Comment by rashmi73** [comment]
@r0hit-gupta  yes was investigating the issue.


@veshu here is detailed explanation for it.
1. It works perfectly well in firefox, issue is just with chrome from the library itself.

2.regarding showing button for shortcuts panel, react-dates by default shows "?" button at bottom right corner of picker, but it is not working in chrome, but working perfectly in firefox.

Also to justify the above comments an issue has been already raised at "react-dates" repo for it.
here is link: https://github.com/airbnb/react-dates/issues/1706

Also for now shall I just remove the message "Press the down arrow key to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates"?
so that reader is not confused on chrome? kindly suggest so I raise PR immediately

**Comment by rashmi73** [comment]
@veshu it should work fine now with this PR. my above comment can be ignored.
PR https://github.com/topcoder-platform/community-app/pull/3265

**Comment by r0hit-gupta** [comment]
@rashmi73 please check https://github.com/topcoder-platform/community-app/pull/3265#issuecomment-524540267

**Comment by rashmi73** [comment]
updating in sometime sorry for delay

**Comment by veshu** [comment]
@rashmi73 Are you going to submit the fix any soon? If not please release the issue in next 3-4 hours. Thanks!

**Comment by rashmi73** [comment]
@veshu will submit it immediately

**Comment by rashmi73** [comment]
@veshu updating PR

**Comment by rashmi73** [comment]
@veshu @r0hit-gupta kindly verify taking my PR and make sure 

1.from node_modules react-dates is deleted
2. run 'npm install'

go to firefox and try focus on calendar using tab, then press "?" however possible on your keyboard, shortcuts will open, 

no need to press tab twice to enter calendar.

kindly confirm

**Comment by veshu** [comment]
@rashmi73 This is not fix. Still when I am text box it is reading same. When I press down arrow or question mark it doesn't work. but If I tab once and move focus to calendar then it will work.

Note: Bug bash is closing on 8PM EST, 5 Sept. If this is not fixed by then it will be closed.

**Comment by rashmi73** [comment]
@veshu shall I add a verification video? because working on my firefox

**Comment by veshu** [comment]
@rashmi73 yes you can. Also this needs to be working on chrome.

**Comment by rashmi73** [comment]
@veshu as clarified in my previous comment with a issue link pointed at react-dates, which clearly says lib itself has bug which does not show question mark on chrome, till they fix it i dont think it will work on chrome as no api to access it from outside.

**Comment by veshu** [comment]
@rashmi73 Thats not the issue here. We can see the help mark even in the chrome see this.
https://take.ms/FMVmQ
The issue is when we use the reader, and reaches the start date/end date input box it says "press down key for ..... and ? for ...." from there if I press this keys it won't work. Not even on Firefox.

But if press tab once then those keys will work.

**Comment by veshu** [comment]
@rashmi73 Simple fix here could be to focus the calendar when the input box is focused.

**Comment by rashmi73** [comment]
@veshu ohhh amazing to see question mark visible on your chrome and not mine, but in just 1 tab it works in my firefox but not in your firefox.


yes I understand the issue that need to press tab twice whereas one tab it should work.
yes tried to focus on it by fetching element by javascript and using focus method but failed.

then observed that even without doing anything it work fine on  1 tab itself. hence told you to clear cache and try it.

so now i will put a video which will clarify and help.

**Comment by rashmi73** [comment]
@veshu here is verification video in which I opened app in firefox , then went to filters on challenge listing ,
then pressing tab i reach start date, now i did not press any tab, and clicked "?" because calendar is open.

so now shortcut appears correctly.

**Comment by rashmi73** [comment]
@veshu it should be working now  PR updated 

[record-20.24.53[05.09.2019].zip](https://github.com/topcoder-platform/community-app/files/3580358/record-20.24.53.05.09.2019.zip)

**Comment by drasticdpk** [comment]
@rashmi73 ,  Now while typing the question mark, the pop up is appear for **Keyboard Shortcuts**. Screed reader should read the text too. (You may refer prod env for reference purpose)
![image](https://user-images.githubusercontent.com/25297859/64416980-ec167c80-d0b7-11e9-977e-bd62a45a5303.png)

**Comment by rashmi73** [comment]
@drasticdpk I do not feel adding aria label attribute to the text wrapping div is in my control.

Can we please consider closing this issue now as requirement of the issue is already met.

@veshu waiting for your comment.

**Comment by veshu** [comment]
@rashmi73 It is existing feature broken. The library itself provides the such labels, but now it is breaking.

**Comment by crazyk07** [comment]
Closing ticket due to shift in priority, will not be able to complete but will still pay for effort rendered.

**Comment by crazyk07** [comment]
Payment task has been updated: https://software.topcoder.com/review/actions/ViewProjectDetails?pid=30099900<br/><br/>```This is an automated message for crazyk via Topcoder X```
